bukkit/src/main/java/com/github/games647/fastlogin/bukkit/task/FloodgateAuthTask.java

```diff
@@ -59,21 +66,42 @@ public void run() {
                     player.getName());
             return;
         }
-        
-        if (!isRegistered && !autoRegisterFloodgate) {
+
+        //decide if checks should be made for conflicting Java player names
+        if (autoLoginFloodgate.equals("no-conflict")
+                || !isRegistered && autoRegisterFloodgate.equals("no-conflict")) {
+            // check for conflicting Premium Java name
+            Optional<Profile> premiumUUID = Optional.empty();
+            try {
+                premiumUUID = plugin.getCore().getResolver().findProfile(username);
+            } catch (IOException | RateLimitException e) {
+                e.printStackTrace();
```

Comment:
The logger is responsible for stacktrace formatting. Otherwise it would end up in stdio. Just use the existing logging message.

Reply:
Rate limiting is a big issue. Try to reduce as much as possible. In this case, couldn't we use the result from the login if we made a check there?

Reply:
`e.printStackTrace();` was auto generated by Eclipse and I forgot to remove it. Will do it.

> couldn't we use the result from the login

If you are refering to 
https://github.com/games647/FastLogin/blob/main/core/src/main/java/com/github/games647/fastlogin/core/shared/JoinManagement.java#L80 then no, because that whole function will be skipped if a player is connecting from Floodgate. After all, Floodgate players will never be true premium Java players.